Framework/MDAlgorithms/src/SmoothMD.cpp

```diff
@@ -126,8 +126,8 @@ KernelVector gaussianKernel(const double fwhm) {
 KernelVector renormaliseKernel(KernelVector kernel,
                                std::vector<bool> validity) {
 
-  if (std::accumulate(validity.cbegin(), validity.cend(), 0.0) <
-      kernel.size()) {
+  if (std::accumulate(validity.cbegin(), validity.cend(),
+                      static_cast<size_t>(0)) < kernel.size()) {
```

Comment:
I *think* one can avoid copying `kernel` again in `normaliseKernel` by passing a reference. I haven't looked carefully enough to see it one could also do the same with `renormaliseKernel` and not sure if enough time is spent in these functions to justify the work.